packages/cli/lib/services/model.service.ts

```diff
@@ -139,6 +139,20 @@ class ModelService {
             if (hasUndefined) {
                 tsType = `${tsType} | undefined`;
             }
+
+            if (tsType.includes('|')) {
+                const types = tsType.split('|');
+                const isEnum = types.every((type) => !JAVASCRIPT_PRIMITIVES.includes(type.trim()));
```

Comment:
I am gonna be a bit pedantic but `|` represents union type not enum

Reply:
are you sure about the negation. Isn't it a union type if every element are primitives?

Reply:
+1 on union. Also the yaml should be an array of string no?
```yaml
gender: 
  - female
  - male
```

Reply:
"Enum" was a misnomer here, but really shorthand for what was trying to be achieved here:

the nango yaml would be this:

```
MyModel:
   gender: male|female
```

generated ts file
```
  gender: 'male' | 'female'
```

So just a string literal type. 

> Isn't it a union type if every element are primitives?
I'm not sure what you mean here? In this case we only want to characterize this as a string literal if it doesn't contain any primitives

> Also the yaml should be an array of string no?
When translated to typescript that wouldn't give us exactly what we want, but maybe I'm misunderstanding what you mean here?

Reply:
enums in json schema are array of possibles values, so I was thinking it could work if we do the same in yaml
https://json-schema.org/understanding-json-schema/reference/enum

avoid false positive and parsable natively